src/app/components/collection/collection.component.ts

```diff
@@ -0,0 +1,239 @@
+        if (isOpenID) {
+            this.authentMode = 'openid';
+        }
+        if (isIam) {
+            this.authentMode = 'iam';
+        }
+    }
+
+    public compareFn(o1, o2): boolean {
+        return o1 && o2 ? o1.name === o2.name : o1 === o2;
+    }
+
+    public ngOnInit(): void {
+        if (this.arlasStartupService.shouldRunApp) {
+            if (!this.isAuthentActivated) {
+                this.getCollections();
+                this.displayedColumns.push('action');
+            } else if (this.authentMode === 'iam') {
+                if (this.arlasIamService.user) {
+                    this.displayedColumns.push(...['is_public', 'owner', 'shared_with']);
+                    this.getCollectionsByOrg();
+                    this.connected.set(true);
+                    this.collectionService.setOptions({
+                        headers: {
+                            Authorization: 'bearer ' + this.arlasIamService.getAccessToken()
+                        }
+                    });
+                } else {
+                    this.organisations.set([]);
+                    this.getCollections();
+                    this.connected.set(false);
+                    this.collectionService.setOptions({});
+                }
+            } else {
+                this.organisations.set([]);
+                this.getCollections();
+                this.connected.set(false);
+                this.collectionService.setOptions({});
+            }
+        }
+    }
+
+    public ngAfterViewInit(): void {
+        this.collectionDataSource.paginator = this.paginator;
+    }
+
+    public getCollectionsByOrg() {
+        this.isLoading = true;
+        this.collections.set([]);
+        const iamHeader = {
+            Authorization: 'Bearer ' + this.arlasIamService.getAccessToken()
+        };
+        const fetchOptions = { headers: iamHeader };
+        this.collabSearchService.setFetchOptions(fetchOptions);
+        this.collectionService.getCollectionsReferenceDescription()
+            .pipe(finalize(() => this.isLoading = false))
+            .subscribe({
+                next: (crds) => {
+                    const collectionsList: CollectionInfos[] = [];
+                    crds.forEach((clc: CollectionInfos) => {
+                        collectionsList.push(clc);
+                    });
+                    // remove owner from shared organisations
+                    collectionsList
+                        .forEach(c => c.display_shared_orgs = c.params.organisations.shared.filter(o => o !== c.params.organisations.owner));
+                    this.collections.set(collectionsList);
+                    this.organisations.set(Array.from(new Set(collectionsList.map(c => c.params.organisations.owner))));
+                    this.organisationsNames.set(this.organisations());
+                    this.collectionDataSource.data = this.collections();
+                }
```

Comment:
what happens in case an error occurs ? shouldn't we empty the collectionsList and prompt an error message ?

Reply:
`this.collections` is already init to empty
I will add a message for the error case